src/test/java/org/pesmypetcare/webservice/controller/petmanager/PetMedicationRestControllerTest.java

```diff
@@ -0,0 +1,135 @@
+        field = "quantity";
+        value = 3;
+        urlBase = "/medication";
+    }
+
+    @Test
+    public void createMedicationShouldReturnStatusOk() throws Exception {
+        willDoNothing().given(service).createMedication(anyString(), anyString(), anyString(), isA(MedicationEntity.class));
+        mockMvc.perform(post(urlBase + "/" + owner + "/" + petName + "/" + dateName)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonMedicationEntity))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void deleteByDateAndNameShouldReturnStatusOk() throws Exception {
+        willDoNothing().given(service).deleteByDateAndName(anyString(), anyString(), anyString());
+        mockMvc.perform(delete(urlBase + "/" + owner + "/" + petName + "/" + dateName)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void deleteAllMedicationsShouldReturnStatusOk() throws Exception {
+        willDoNothing().given(service).deleteAllMedications(anyString(), anyString());
+        mockMvc.perform(delete(urlBase + "/" + owner + "/" + petName))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void getMedicationDataShouldReturnMedicationEntityAndStatusOk() throws Exception {
+        willReturn(MedicationEntity).given(service).getMedicationData(anyString(), anyString(), anyString());
+        mockMvc.perform(get(urlBase + "/" + owner + "/" + petName + "/" + dateName))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void getAllMedicationDataShouldReturnMedicationEntityListAndStatusOk() throws Exception {
+        willReturn(MedicationEntityList).given(service).getAllMedicationData(anyString(), anyString());
+        mockMvc.perform(get(urlBase + "/" + owner + "/" + petName))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void getAllMedicationsBetweenShouldReturnMedicationEntityListAndStatusOk() throws Exception {
+        willReturn(MedicationEntityList).given(service).getAllMedicationsBetween(anyString(), anyString(), anyString(),
+                anyString());
+        mockMvc.perform(get(urlBase + "/" + owner + "/" + petName + "/between/" + date + "/" + date2))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void getMedicationFieldShouldReturnFieldValueAndStatusOk() throws Exception {
+        willReturn(value).given(service).getMedicationField(anyString(), anyString(), anyString(), anyString());
+        mockMvc.perform(get(urlBase + "/" + owner + "/" + petName + "/" + dateName + "/" + field))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void updateMedicationFieldShouldReturnStatusOk() throws Exception {
+        willDoNothing().given(service).updateMedicationField(anyString(), anyString(), anyString(), anyString(), anyString());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 126).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)